airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/BigQueryDestinationAcceptanceTest.java

```diff
@@ -306,26 +304,22 @@ public boolean requiresDateTimeConversionForNormalizedSync() {
 
   @Override
   public void convertDateTime(ObjectNode data, Map<String, String> dateTimeFieldNames) {
-    var fields = StreamSupport.stream(Spliterators.spliteratorUnknownSize(data.fields(),
-        Spliterator.ORDERED), false).toList();
     if (dateTimeFieldNames.keySet().isEmpty()) {
       return;
     }
-    fields.forEach(field -> {
-      for (String path : dateTimeFieldNames.keySet()) {
-        var key = field.getKey();
-        if (isKeyInPath(path, key) && DateTimeUtils.isDateTimeValue(field.getValue().asText())) {
-          switch (dateTimeFieldNames.get(path)) {
-            case DATE_TIME -> {
-              var result = String.valueOf(new BigDecimal(DateTimeUtils.getEpochMicros(field.getValue().asText())).divide(new BigDecimal(1000000)));
-              data.put(key.toLowerCase(), !result.contains(".") ? result + ".0" : result);
-            }
-            case DATE -> data.put(key.toLowerCase(),
-                DateTimeUtils.convertToDateFormat(field.getValue().asText()));
+    for (String path : dateTimeFieldNames.keySet()) {
+      if (isOneLevelPath(path) && !data.at(path).isMissingNode() && DateTimeUtils.isDateTimeValue(data.at(path).asText())) {
```

Comment:
Could this condition just be `if (!data.at(path).isMissingNode())`?
* Why do we need to check that this is a one-level path? Shouldn't we do this replacement even for deeply-nested values?
* Are we not guaranteed that this path contains a date/time value?

Reply:
> Why do we need to check that this is a one-level path? Shouldn't we do this replacement even for deeply-nested values?

There are two approaches to implementation for convertDateTime(..):
1. The first approach (like the method above) is used for connectors, which serialize deeply-nested values as jsonb string and there is no conversion for nested date-time fields (Postgres, MSSQL, Redshift, etc). It uses `isOneLevelPath()` method to convert only one-level fields without nested ones. 
2. The second approach is used for connectors that have a conversion for deeply-nested date-time values (GcsAvro, S3Parquet, BigQueryDenormalizedGcs etc.) 

> Are we not guaranteed that this path contains a date/time value?

I added this verification to be sure cases when there are two identical paths with the different formats will be processed in the right way:
```
message1:
{
  "details": {
    "test_field": "2022-01-23"
  }
}
message2:
{
  "details": {
    "test_field": "test string value"
  }
}
```

Reply:
ahhh, got it. makes sense, thanks!